src/sklearn_raster/estimator.py

```diff
@@ -444,9 +442,170 @@ def kneighbors(
             **kneighbors_kwargs,
         )
 
+    @require_implementation
+    @require_fitted
+    @require_attributes("get_feature_names_out")
+    def transform(
+        self,
+        X: FeatureArrayType,
+        *,
+        skip_nodata: bool = True,
+        nodata_input: NoDataType = None,
+        nodata_output: float | int = np.nan,
+        ensure_min_samples: int = 1,
+        allow_cast: bool = False,
+        check_output_for_nodata: bool = True,
+        **transform_kwargs,
+    ) -> FeatureArrayType:
+        """
+        Apply the transformation to n-dimensional X features.
+
+        Parameters
+        ----------
+        X : Numpy or Xarray features
+            The n-dimensional input features. Array types should be in the shape
+            (features, ...) while xr.Dataset should include features as variables.
+            Features should correspond with those used to fit the estimator.
+        skip_nodata : bool, default=True
+            If True, NoData and NaN values will be skipped during prediction. This
+            speeds up processing of partially masked features, but may be incompatible
+            if estimators expect a consistent number of input samples.
+        nodata_input : float or sequence of floats, optional
+            NoData values other than NaN to mask in the output features. A single value
+            will be broadcast to all features while sequences of values will be assigned
+            feature-wise. If None, values will be inferred if possible based on
+            available metadata.
+        nodata_output : float or int or tuple, optional
+            NoData samples in the input features will be replaced with this value in the
+            output features. If the value does not fit the array dtype returned by the
+            estimator, an error will be raised unless `allow_cast` is True. Defaults to
+            np.nan.
+        ensure_min_samples : int, default 1
+            The minimum number of samples that should be passed to `transform`. If the
+            array is fully masked and `skip_nodata=True`, dummy values (0) will be
+            inserted to ensure this number of samples. The minimum supported number of
+            samples depends on the estimator used. No effect if the array contains
+            enough unmasked samples or if `skip_nodata=False`.
+        allow_cast : bool, default=False
+            If True and the estimator output dtype is incompatible with the chosen
+            `nodata_output` value, the output will be cast to the correct dtype instead
+            of raising an error.
+        check_output_for_nodata : bool, default True
+            If True and `nodata_output` is not np.nan, a warning will be raised if the
+            selected `nodata_output` value is returned by the estimator, as this may
+            indicate a valid sample being masked.
+        **transform_kwargs
+            Additional arguments passed to the estimator's `transform` method.
```

Comment:
I was comparing this to the other methods' docstrings when reviewing (`predict`, `predict_proba`, `kneighbors`).  In this line, you've backticked `transform`, whereas the others don't have it.  I like the back-ticking, so I'd apply the same pattern to the other docstrings.